elastic/apm/component-template.json

```diff
@@ -0,0 +1,10 @@
+{
+  "template": {
+    "settings": {
+      "index": {
+        "number_of_shards": {{number_of_shards | default(1)}},
+        "number_of_replicas": {{number_of_replicas | default(0)}}
+      }
+    }
+  }
+}
```

Comment:
The [index templates from the ES apm plugin](https://github.com/elastic/elasticsearch/tree/main/x-pack/plugin/apm-data/src/main/resources/index-templates) already contain a `composed_of` section; a common one is `apm@settings`. 
If this component template was named `apm@settings`, then it should be included in the default index templates.

Reply:
Thanks, I was confused with the one we had in the integration package. Will name this as apm@settings.

Reply:
based on your explanations, let's keep things as you have them (see comment below)